zebra-state/src/service/chain_tip.rs

```diff
@@ -320,6 +321,16 @@ impl ChainTipChange {
         Ok(action)
     }
 
+    /// Returns:
+    /// - `Some(TipAction)` if there has been a change since the last time the method was called.
+    /// - [`None`] if there has been no change.
+    pub fn get_tip_change(&mut self) -> Option<TipAction> {
+        match self.tip_change().now_or_never().transpose() {
```

Comment:
### Tricky tokio edge case

We'll need to disable tokio's cooperative scheduling, otherwise we will get spurious `None`s from `now_or_never`.

(The warning about starvation doesn't apply, since we're using `now_or_never`, which always returns immediately.)

https://docs.rs/tokio/1.11.0/tokio/task/index.html#cooperative-scheduling

I'll open a PR to update the [async RFC](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/rfcs/0011-async-rust-in-zebra.md). This seems like an edge case we'll need to remember.

```suggestion
        match tokio::task::unconstrained(self.tip_change()).now_or_never().transpose() {
```

Reply:
I just found out that `tokio::task::unconstrained` is new in tokio 1.12, but we're only using tokio 0.3.7.

Here's the documentation for 0.3.7 - there's no `unconstrained`, but there is preemption:
https://docs.rs/tokio/0.3.7/tokio/task/index.html#what-are-tasks
https://tokio.rs/blog/2020-04-preemption

So I think it might be better to call `ChainTipChange::action` directly from `last_tip_change`.

You can get the last block using [watch::Receiver::borrow](https://docs.rs/tokio/0.3.7/tokio/sync/watch/struct.Receiver.html#method.borrow).

Here are some other things to be aware of during the refactor.
(They're not documented in the code - sorry about that!)
* the `assert!` only applies to `wait_for_tip_change` - if the last hash is the same, that's when `last_tip_change` will want to return `None`
* the `last_change_hash` update needs to be done for both `wait_for_tip_change` and `last_tip_change`